src/SprykerEco/Yves/CrefoPay/Form/DataProvider/BillFormDataProvider.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\CrefoPay\Form\DataProvider;
+
+use Generated\Shared\Transfer\CrefoPayPaymentTransfer;
+use Generated\Shared\Transfer\PaymentTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Shared\Kernel\Transfer\AbstractTransfer;
+use Spryker\Yves\StepEngine\Dependency\Form\StepEngineFormDataProviderInterface;
+use SprykerEco\Yves\CrefoPay\CrefoPayConfig;
+use SprykerEco\Yves\CrefoPay\Form\BillSubForm;
+
+class BillFormDataProvider implements StepEngineFormDataProviderInterface
+{
+    /**
+     * @var \SprykerEco\Yves\CrefoPay\CrefoPayConfig
+     */
+    protected $config;
+
+    /**
+     * @param \SprykerEco\Yves\CrefoPay\CrefoPayConfig $config
+     */
+    public function __construct(CrefoPayConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function getData(AbstractTransfer $quoteTransfer): QuoteTransfer
+    {
+        if ($quoteTransfer->getPayment() === null) {
```

Comment:
I understand that it's highly unlikely that you have here not QuoteTransfer, but just in case I would prefer that you have some sort of check here.

Reply:
I think it's a core issue. They should provide proper typehint but they don't want to because it leads to major. I hope core fix it in nearest future. I would prefer to keep it as is and we won't have unnecessary ridiculous check after strict declaration.

Reply:
But in case it will be fixed in a core, we will have to change this method declaration anyway. And in this case, remove some unnecessary check is not so hard IMO